src/components/PostDetails.tsx

```diff
@@ -5,57 +5,36 @@ import { NewCommentForm } from './NewCommentForm';
 import * as commentsApi from '../api/comments';
 
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { actions as commentsACtions } from '../features/comments/commentsSlice';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const {
+    comments,
+    loaded,
+    hasError,
+  } = useAppSelector(state => state.comments);
   const [visible, setVisible] = useState(false);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    dispatch(commentsACtions.setIsLoaded(false));
+    dispatch(commentsACtions.setHasError(false));
     setVisible(false);
 
     commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+      .then((response) => dispatch(commentsACtions.setComments(response)))
```

Comment:
I noticed that you use 2 different approaches in the project `async/await` and `then` for working with async functions. It's generally recommended to be consistent and use a single approach throughout a project